### PR TITLE
Make transaction api use cache after node restart

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview

Problem:
THe node would get the transactions from scratch after restarting even the data is stored in the database.

The pr is intended to fix this. The node should retrieve the data directly from database after node restarting.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Review by nzpr — APPROVED** [review]
LGTM

File: node/src/main/scala/coop/rchain/node/web/Transaction.scala [hunk 1]
```diff
@@ -1,8 +1,9 @@
 package coop.rchain.node.web
 
 import cats.effect.concurrent.Deferred
-import cats.effect.{Concurrent, Sync}
+import cats.effect.Concurrent
 import cats.syntax.all._
+import cats.instances.option
```
Comment (tgrospic) [inline node/src/main/scala/coop/rchain/node/web/Transaction.scala]: Is this option instance necessary?

**Review by tgrospic — APPROVED** [review]
LGTM

File: node/src/main/scala/coop/rchain/node/web/Transaction.scala (hunk 1 above)
Comment (zsluedem) [reply node/src/main/scala/coop/rchain/node/web/Transaction.scala]: Good catch

**Comment by zsluedem** [comment]
bors merge

**Comment by zsluedem** [comment]
bors merge

**Pull request merged**